packages/angular_devkit/build_optimizer/src/transforms/scrub-file.ts

```diff
@@ -427,7 +428,21 @@ function pickDecorateNodesToRemove(
 
       return true;
     });
-    ngDecoratorCalls.push(...metadataCalls);
+    // Remove all __param calls.
+    const paramCalls = elements.filter((el) => {
+      if (!isTslibHelper(el, '__param', tslibImports, checker)) {
+        return false;
+      }
+      if (el.arguments.length != 2) {
+        return false;
+      }
+      if (el.arguments[0].kind !== ts.SyntaxKind.NumericLiteral) {
+        return false;
+      }
```

Comment:
why do you think it's important to have the length and arguments[0] check? do you think we could misidentify the __params helper somehow?

Reply:
Throughout Build Optimizer we try to keep these checks tight. Sometimes all we can say is that something has the right identifier text, and there are some situations where it can go wrong. 

Since Build Optimizer will end up processing a lot of code we try to err on the side of caution. Here the best we can say is "it's a call expression called __param with two arguments, where the first one one is a number". 

I think it's very unlikely to reach this section of the code if we're not in the happy path, but if we remove something that we shouldn't then it's very likely the app breaks.

Reply:
sgtm